### ISSUE TITLE
Panicked if type(self)() in __del__ methods 

### ISSUE THREAD

**Issue by xiaxinmeng** [RustPython/RustPython#4863]
This code creates an instance of the Dummy class and then immediately deletes it, which triggers the __del__ method to create a new instance of the Dummy class. 

```
class Dummy:
   def __del__(self):
      type(self)()

d = Dummy()
del d
```

Crash message:
```
Exception ignored in: <object function repr() failed>
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: PyBaseException', vm/src/stdlib/sys.rs:593:59
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Environment:
Ubuntu 18.04
rustpython v0.2.0

**Comment by itsankitkp** [comment]
It gets stuck in cpython (3.12a)

**Comment by DimitrisJim** [comment]
we should probably hang too? This will continuously try and call `del` since it's creating a new `d` and then immediately discarding it, triggering a loop.

The call to `unwrap` definitely needs fixing though.

**Comment by youknowone** [comment]
In general, Panic is better than hang. Both are useless but hanging is silent.

**Comment by fanninpm** [comment]
@xiaxinmeng What is the intended result in CPython? This seems like you want the interpreter to hang.

**Comment by DimitrisJim** [comment]
Yeah, the correct behaviour here is to hang due to this continuously calling `__del__`.

**Comment by youknowone** [comment]
reported as fixed by @changjoon-park
